src/main/java/at/hannibal2/skyhanni/features/inventory/SuperCraftFeatures.kt

```diff
@@ -0,0 +1,31 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.GetFromSackAPI
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.utils.DelayedRun
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.PrimitiveItemStack
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class SuperCraftFeatures {
+    private val craftedPattern by RepoPattern.pattern(
+        "inventory.supercrafting.craft",
+        "§eYou Supercrafted §r§r§r§[^.*]([^x§]+?)(?:§r§8x(\\d+))?§r§e!"
```

Comment:
Cleanup of the regex.
First idk what [^.*] should mean, it simply can be a . for any character
Second do not exclude x from the name of the item, makes problems when an item has an x in the name. The exclusion of the colorcode does the job anyway.
Third do not make the search lazy (+? / *? ) without needing it since most of the time it *doubles* the runtime of the regex matching (according to regex101 and php flavor (since it is the only one that counts the steps), I know that isn't acurade but still I image that the loss is still no insignificance).
```suggestion
        "§eYou Supercrafted §r§r§r§.([^§]+)(?:§r§8x(\\d+))?§r§e!"
```

Reply:
To add onto this please use name capture groups, in all of skyhanni there are 6 places where we use unnamed capture groups (which I dont like) and 238 where we use named ones

Reply:
Thx, for making me aware that we only use named capturing groups (Now I go an change about 50+ regex, maybe, in my open PRs)
```suggestion
        "§eYou Supercrafted §r§r§r§.(?<item>[^§]+)(?:§r§8x(?<amount>\\d+))?§r§e!"
```

Reply:
alright will make sure to use named groups next time

Reply:
also i dont even rememer exculding "x" lol sorry